op-service/sources/l1_beacon_client.go

```diff
@@ -223,7 +223,7 @@ func (cl *L1BeaconClient) fetchSidecars(ctx context.Context, slot uint64, hashes
 	for i := 0; i < cl.pool.Len(); i++ {
 		f := cl.pool.Get()
 		resp, err := f.BeaconBlobSideCars(ctx, cl.cfg.FetchAllSidecars, slot, hashes)
-		if err != nil {
+		if err != nil || len(resp.Data) == 0 {
```

Comment:
Isn't this already caught by `if len(hashes) != len(apiscs) {` in `GetBlobSidecars`?

Reply:
Right that catches it later which is not salient here because we want to go through the fallback clients. We check it here so we actually do go through the beacon clients available

Reply:
hmm I saw a response by someone, but it got deleted again? Anyways, wouldn't this check
```go
	for _, apisc := range resp.Data {
		delete(indices, uint64(apisc.Index))
	}
	if len(indices) > 0 {
		return eth.APIGetBlobSidecarsResponse{}, fmt.Errorf("#returned blobs(%d) != #requested blobs(%d)", len(hashes)-len(indices), len(hashes))
	}
```
in the `BeaconHTTPClient.BeaconBlobSideCars` implementation already guarantee that we return an error in this case?

Reply:
No an error wasn't returned - I received nil error & trivial APIGetBlobSidecarsResponse empty struct.

Reply:
Couldn't that just mean there were no blobs in that slot?

Reply:
right , no blobs at that slot because the otherside has it expired and doesn't have it anymore. Hence we should try through the other blob providers (beacon chains) we have.

Reply:
I provided my local prysm node as the l1.beacon and then other beacon archivers as the fallback. If you say that its legitimate for no blobs given on that slot because the l1.beacon expired the blobs, then how would the fallbacks ever get picked if this is giving back no error and never goes to the next beacon clients?

Reply:
It's a known issue of Beacon clients that they just return an empty blobs list and no error, if the blobs expired on their end @tynes 

@fxfactorial What I'm getting at is that the `BeaconHTTPClient.BeaconBlobSideCars` implementation already checks that the beacon response contains all the requested hashes, that's what the code
```
	indices := make(map[uint64]struct{}, len(hashes))
	for _, h := range hashes {
		indices[h.Index] = struct{}{}
	}
	for _, apisc := range resp.Data {
		delete(indices, uint64(apisc.Index))
	}
	if len(indices) > 0 {
		return eth.APIGetBlobSidecarsResponse{}, fmt.Errorf("#returned blobs(%d) != #requested blobs(%d)", len(hashes)-len(indices), len(hashes))
	}
```
does. Why doesn't this catch the case that a Beacon endpoint returns an empty list of blobs? Wouldn't this then cause this code to return an error?

Would be great to add a regression test to this PR that fails without your change. That would show which scenario wasn't covered before.